ktor-server/ktor-server-core/common/src/io/ktor/server/routing/RoutingNode.kt

```diff
@@ -350,6 +349,38 @@ private fun RoutingNode.getAllRoutes(endpoints: MutableList<RoutingNode>) {
     children.forEach { it.getAllRoutes(endpoints) }
 }
 
+/**
+ * String representation of the path matched by this route.
+ *
+ * [Report a problem](https://ktor.io/feedback/?fqname=io.ktor.server.routing.RoutingNode.path)
+ */
+public val RoutingNode.path: String
+    get() = path()
+
+private fun RoutingNode.path(): String {
+    val parentPath = parent?.path()
+    val selectorElement = selector.toPathElement()
+    return when {
+        parentPath == null -> selectorElement
+        selectorElement.isEmpty() -> parentPath
+        parentPath.endsWith('/') || selectorElement.startsWith('/') -> "$parentPath$selectorElement"
+        else -> "$parentPath/$selectorElement"
+    }
+}
+
+private fun RouteSelector.toPathElement(): String = when (this) {
+    is PathSegmentConstantRouteSelector,
+    is PathSegmentParameterRouteSelector,
+    is PathSegmentOptionalParameterRouteSelector,
+    is PathSegmentTailcardRouteSelector,
+    is PathSegmentWildcardRouteSelector,
+    is PathSegmentRegexRouteSelector -> toString()
```

Comment:
We could introduce a supertype `PathSegmentRouteSelector` for these selectors to make it possible to create custom route selectors appearing in path.
@bjhham, what do you think?